src/main/java/sort/NaturalSort.java

```diff
@@ -0,0 +1,37 @@
+package sort;
+
+import org.neo4j.procedure.Name;
+import org.neo4j.graphdb.Node;
+import org.neo4j.procedure.Procedure;
+import org.neo4j.procedure.UserFunction;
+import sort.*;
+import java.util.*;
+
+public class NaturalSort
+{
+    @UserFunction
+    // @Description("sort.natural_list(coll) sort on primitive collection")
+    public List<Object> natural_list(
+        @Name("coll") List<Object> coll
+        ) {
+	    if (coll == null || coll.isEmpty()) return Collections.emptyList();
+        List<Object> sorted = new ArrayList<>(coll);
+        Collections.sort(sorted, new NaturalOrderComparator());
+        return sorted;
+    }
+
+    @UserFunction
+    // @Description("sort.natural_obj(coll, props) sort on collection of nodes and specific props")
+    public List<Node> natural_obj(
+        @Name("coll") List<Node> coll,
+        @Name("props") String props
+        ) {
+	    if (props == null) return coll;
+	    if (coll == null || coll.isEmpty()) return Collections.emptyList();
+        List<Node> sorted = new ArrayList<>(coll);
+        NaturalOrderNodeComparator comparator = new NaturalOrderNodeComparator();
+        comparator.setProps(props); // This does not work yet, we should set this props somehow
```

Comment:
this method not working? or all sorting?

Reply:
It works for me without changes. My test:
```java
    @Test
    public void shouldReturnSortedValuesByProp() {
        try( Driver driver = GraphDatabase.driver(embeddedDatabaseServer.boltURI(), driverConfig);
             Session session = driver.session()) {
            session.run("CREATE (o:node{name: 'n1', rfd: '=A2'})");
            session.run("CREATE (o:node{name: 'n2', rfd: '=A10'})");
            session.run("CREATE (o:node{name: 'n3', rfd: '=A1'})");
            String result = session.run(
                "MATCH (o:node)\n" +
                "RETURN [n IN sort.natural_obj(collect(o), 'rfd') | n.rfd ] AS result"
            ).single().get("result").toString();
            assertThat(result).isEqualTo(( "[\"=A1\", \"=A2\", \"=A10\"]" ));
        }
    }
```

Reply:
Holy shit! I've realized that I've changed the version of Maven and have some changes in the environment and that was probably the reason why it won't work. Thanks @rossinek ❤️